#!/usr/bin/env python
PACKAGE = "tuw_iwos_odometer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("linear_velocity_tolerance", double_t, 0, "tolerance for linear velocity matching", 0.5, 0.0, 1.0)
gen.add("angular_velocity_tolerance", double_t, 0, "tolerance for angular velocity matching", 0.3, 0.0, 1.0)
gen.add("steering_position_tolerance", double_t, 0, "tolerance for steering position matching", 0.05, 0.0, 1.0)

gen.add("calculation_iteration_duration", double_t, 0, "number of odometer calculation iterations", 0.001, 0, 0.1)

exit(gen.generate(PACKAGE, "tuw_iwos_odometer", "OdometerServiceNode"))
